crates/floresta-wire/src/p2p_wire/sync_node.rs

```diff
@@ -70,6 +79,15 @@ where
         try_and_log!(self.request_blocks(blocks).await);
     }
 
+    /// Starts the sync node by updating the last block requested and starting the main loop.
+    /// This loop to the following tasks, in order:
+    ///     - Receives messages from the node_rx channel.
+    ///     - Handles the message received.
+    ///     - Checks if the kill signal is set, if so, breaks the loop.
+    ///     - Checks if the chain is in IBD and disables it if it's not (e.g. if the chain is synced).
+    ///     - Checks if our tip is obsolete and requests a new one, creating a new connection.
```

Comment:
I don't think this node checks for old tip

Reply:
It does.

```Rust
            if Instant::now()
                .duration_since(self.0.last_tip_update)
                .as_secs()
                > SyncNode::ASSUME_STALE
            {
                self.1.last_block_requested = self.chain.get_validation_index().unwrap();
                self.create_connection(ConnectionKind::Regular).await;
                self.last_tip_update = Instant::now();
                continue;
            }

```

Reply:
Ok, I was thinking something else. But I think the doc i accurate anyway